common/test/acceptance/pages/studio/settings_advanced.py

```diff
@@ -34,6 +37,47 @@ def _is_finished_loading():
         EmptyPromise(_is_finished_loading, 'Finished rendering the advanced policy items.').fulfill()
         return self.q(css='body.advanced').present
 
+    def get_key_names(self):
+        """
+            Returns a list of key names of all settings.
+        """
+        return self.q(css=KEY_CSS).text
+
+    def get_deprecated_settings_button_text(self):
+        """
+            Returns text for deprecated settings button
+        """
+        return self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).text[0]
+
+    def click_deprecated_settings_button(self):
+        """
+            Show deprecated Settings
+        """
+        if self.get_deprecated_settings_button_text() == 'Show Deprecated Settings':
+            self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).click()
+            self.wait_for_element_presence(
+                DEPRECATED_SETTINGS_SELECTOR,
+                'Deprecated Settings are not present'
+            )
+        else:
+            self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).click()
+            self.wait_for_element_absence(
+                DEPRECATED_SETTINGS_SELECTOR,
+                'Deprecated Settings are present'
+            )
```

Comment:
It seems this Page Object method is performing a validation than just clicking on deprecated settings button, so instead of checking the scenario that application is behaving correctly after the click in the test, we are performing these checks behind the scenes on page object. Can we move the condition to test?
On a different note if it is unavoidable, change the code to 
`
self.q(css=DEPRECATED_SETTINGS_BUTTON_SELECTOR).click()
if self.get_deprecated_settings_button_text() == 'Show Deprecated Settings':
    self.wait_for_element_presence(
        DEPRECATED_SETTINGS_SELECTOR,
        'Deprecated Settings are not present'
    )
else:
self.wait_for_element_absence(
        DEPRECATED_SETTINGS_SELECTOR,
        'Deprecated Settings are present'
    )
`